sway-core/tests/ir_to_asm/ret_string_in_struct.asm

```diff
@@ -0,0 +1,37 @@
+.program:
+ji   i4
+noop
+DATA_SECTION_OFFSET[0..32]
+DATA_SECTION_OFFSET[32..64]
+lw   $ds $is 1
+add  $$ds $$ds $is
+lw   $r1 $fp i73              ; load input function selector
+lw   $r0 data_4               ; load fn selector for comparison
+eq   $r0 $r1 $r0              ; function selector comparison
+jnzi $r0 i14                  ; jump to selected function
+lw   $r0 data_5               ; load fn selector for comparison
+eq   $r0 $r1 $r0              ; function selector comparison
+jnzi $r0 i22                  ; jump to selected function
+rvrt $zero                    ; revert if no selectors matched
+move $r2 $sp                  ; save register for temporary stack value
+cfei i8                       ; allocate 8 bytes for temporary struct
+lw   $r1 data_0               ; literal instantiation
```

Comment:
I'm surpised this works.  I would've thought it's loading `"foob"` into `$r1`.

Reply:
This is similar to what we do with `b256` I believe. You basically just get back the pointer.